diff --git a/build/config/clang/clang.gni b/build/config/clang/clang.gni
index 1aad3d60b8404..c600c0edf481c 100644
--- a/build/config/clang/clang.gni
+++ b/build/config/clang/clang.gni
@@ -19,4 +19,7 @@ declare_args() {
       ((is_linux && !is_castos) || (is_android && !is_cast_android))
 
   clang_base_path = default_clang_base_path
+
+  # Specifies whether or not bitcode should be embedded in all compiled targets
+  clang_embit_bitcode = false
 }
diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index e89b62eb0aaf6..58bcaac2ffbf8 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -847,6 +847,10 @@ config("compiler") {
     }
   }
 
+  if (is_clang && clang_embed_bitcode) {
+    cflags += [ "-Xclang", "-fembed-bitcode=all" ]
+  }
+
   # Pass the same C/C++ flags to the objective C/C++ compiler.
   cflags_objc += cflags_c
   cflags_objcc += cflags_cc
